### PR TITLE
Change asset path filter to use prefix search

### LINKED ISSUE #77: Change asset `path` query param to match using prefix
ATM there is only 
```
/dandisets/{parent_lookup_version__dandiset__pk}/versions/{parent_lookup_version__version}/assets/
```
to get listing of all the assets in a dandiset.  If it is desired to download just a single folder (subject), it is needed to first get a full listing (could be large) and then select only the entries matching the desired location. There is also
```
/dandisets/{parent_lookup_version__dandiset__pk}/versions/{parent_lookup_version__version}/assets/paths/
```
to just get paths on the top level of the dandiset.  ATM we do have indeed only 1 level of subdirectories, but it will change e.g. with BIDS (could have `ses-XXX/` subfolder per subject).

I think the easiest way around would be to allow `location=` option (mimicing current new web UI file browsing option name, as in https://deploy-preview-341--gui-dandiarchive-org.netlify.app/#/dandiset/000006/0.200714.1807/files?location=%2Fsub-anm369962%2F) to be allowed to both of those API calls to limit for a given location

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Fixes #77

### CODE REVIEW

**Comment by brianhelba** [comment]
@AlmightyYakob Can you rebase this on the latest master?

Can you also add a test for this? Let me know if you'd like to discuss what an elegant one would be. I haven't thought about testing filters much yet.

File: dandi/publish/tests/test_asset.py [hunk 1]
```diff
@@ -73,3 +73,44 @@ def test_asset_rest_retrieve(api_client, asset):
...
+        {'path': asset.path[: int(len(asset.path) / 2)]},
+    ).data['results']
+
+    full_path_assets = api_client.get(
+        f'/api/dandisets/{asset.version.dandiset.identifier}/'
+        f'versions/{asset.version.version}/assets/',
+        {'path': asset.path},
+    ).data['results']
+
+    assert asset_data in partial_path_assets
```
Comment (brianhelba) [inline dandi/publish/tests/test_asset.py:115]: Is it possible to reduce what is being asserted, allowing us to remove the giant `asset_data` variable? I think just asserting that the `asset.id` was in the returned data is adequate.

File: dandi/publish/tests/test_asset.py [hunk 2]
```diff
@@ -73,3 +73,44 @@ def test_asset_rest_retrieve(api_client, asset):
...
+        'uuid': str(asset.uuid),
+        'path': asset.path,
+        'size': asset.size,
+        'sha256': asset.sha256,
+        'created': TIMESTAMP_RE,
+        'updated': TIMESTAMP_RE,
+        'metadata': asset.metadata,
+    }
+
+    partial_path_assets = api_client.get(
```
Comment (brianhelba) [inline dandi/publish/tests/test_asset.py:103]: `partial_path_assets` and `full_path_assets` are separate cases and should be in separate functions. This should be less burdensome if we eliminate `asset_data`.

File: dandi/publish/views/asset.py [hunk 3]
```diff
@@ -31,6 +31,8 @@ class Meta:
 
 
 class AssetFilter(filters.FilterSet):
+    path = filters.CharFilter(lookup_expr='istartswith', field_name='path')
+
     class Meta:
         model = Asset
```
Comment (brianhelba) [inline dandi/publish/views/asset.py:37]: You shouldn't need to specify `field_name` if it's the same as the filter name: https://django-filter.readthedocs.io/en/master/guide/usage.html#the-filter

File: dandi/publish/tests/test_asset.py [hunk 4]
```diff
@@ -73,3 +73,44 @@ def test_asset_rest_retrieve(api_client, asset):
         'updated': TIMESTAMP_RE,
         'metadata': asset.metadata,
     }
+
+
+@pytest.mark.django_db
+def test_asset_path_filter(api_client, asset):
```
Comment (brianhelba) [inline dandi/publish/tests/test_asset.py:79]: Do you think it's reasonable to include any test cases where the filter doesn't match? How about ones where the case-insensitivity is tested?

Comment (brianhelba) [reply dandi/publish/tests/test_asset.py:79]: I'm hoping these could maybe be parameterized, perhaps with a bool parameter for whether each sub-case passes.

**Review by brianhelba — DISMISSED** [review]

**Comment by AlmightyYakob** [comment]
@brianhelba lmk if the recent changes I've made address your concerns

**Review by dchiquit — APPROVED** [review]

**Pull request merged**